### PR TITLE
Eslint setup on codebase/pipeline

### LINKED ISSUE #9: Add eslint
- [x] setup eslint 
- [x] add eslint in pipelines

### PR DESCRIPTION
**Pull request by Willian1661**
@eerison 
may you take a look and say if there's something wrong? 
i think it's all done with the Eslint on codebase. 
close: #9

### CODE REVIEW

File: .github/workflows/quality.yml [hunk 1]
```diff
@@ -13,6 +13,6 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
-          node-version: '18.x'
+          node-version: '^18.x'
       - run: npm install
       - run: npm test
```
Comment (eerison) [inline .github/workflows/quality.yml]: just add the lint after test, you dont need this new file atm

**Comment by Willian1661** [comment]
i created a new workflow for the lint, it's needed ? or i can still have just one file with jest and lint?

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	"extends": "eslint:recommended",
+	"overrides": [
+	],
+	"parserOptions": {
+		"ecmaVersion": "latest",
+		"sourceType": "module"
+	},
+	"rules": {
+		"no-useless-call": "error",
+        "no-useless-catch": "error",
```
Comment (eerison) [inline .eslintrc.js:14]: where did you get those config?

File: .github/workflows/quality.yml (hunk 1 above)
Comment (eerison) [inline .github/workflows/quality.yml]: create a script entry point like test

and check only src folder

File: .eslintrc.js (hunk 2 above)
Comment (Willian1661) [reply .eslintrc.js:14]: > where did you get those config?
here:
https://levelup.gitconnected.com/getting-started-with-eslint-development-workflow-f59165fa9735
and here:
https://eslint.org/docs/latest/rules/

**Comment by eerison** [comment]
> i created a new workflow for the lint, it's needed ? or i can still have just one file with jest and lint?

let keep it in one file for now!

File: .github/workflows/quality.yml (hunk 1 above)
Comment (eerison) [reply .github/workflows/quality.yml]: please reply into the thread, it's easier to track the messages

**Comment by Willian1661** [comment]
@eerison 
many many many problems 
images

File: .prettierrc [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+trailingComma: "none",
+semi: false,
```
Comment (eerison) [inline .prettierrc:4]: What is it?

File: .github/workflows/quality.yml [hunk 4]
```diff
@@ -13,6 +13,7 @@ jobs:
...
+      - run: npx eslint src/**
```
Comment (eerison) [inline .github/workflows/quality.yml]: I guess you can define it into the script,  I mean this src

File: .prettierrc (hunk 3 above)
Comment (Willian1661) [reply .prettierrc:4]: i created this prettier to be one script entry point
like test

File: .eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+const Prettierrc = require("./prettier.config.js");
+
+module.exports = {
+	"env": {
+		"browser": true,
+		"es2021": true
+	},
```
Comment (eerison) [inline .eslintrc.js:5]: Do u know what each line does here?

File: .prettierrc (hunk 3 above)
Comment (Willian1661) [reply .prettierrc:4]: it's right doing this? this **.prettierrc** it has other rules, like global rules that **.eslintrc** it'll get also the **.prettier**("_eslint:recommended_" & _.prettier_ will run as extends )

File: .eslintrc.js (hunk 5 above)
Comment (Willian1661) [reply .eslintrc.js:5]: the env browser: creates global  browser variables.
and es2021:  adds all ECMAScript 2021 globals 
says here: https://eslint.org/docs/latest/user-guide/configuring/language-options#specifying-environments

**Comment by Willian1661** [comment]
@eerison 
it passed by all the checks and it's working, but i don't know why :p

File: .github/workflows/quality.yml (hunk 4 above)
Comment (Willian1661) [reply .github/workflows/quality.yml]: i defined inside of overrides:
"overrides": [
		{
		"files": [ "src/**/*.js" ],
	}
		],

File: .eslintrc.js (hunk 5 above)
Comment (eerison) [reply .eslintrc.js:5]: > the env browser: creates global browser variables. and es2021: adds all ECMAScript 2021 globals says here: https://eslint.org/docs/latest/user-guide/configuring/language-options#specifying-environments

Ok, why do you need this?

Comment (eerison) [reply .eslintrc.js:5]: Ok, it's fine! I checked the docs 

But I hope you know what every single rule does 😉

File: .eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+			"double"
+		],
+		"semi": "error",
+        "semi-spacing": "error",
+		"semi-style": [
+            "error",
+            "last"
+        ]
+	}
+}
```
Comment (eerison) [inline .eslintrc.js:46]: @Willian1661 

Could you add those rulus please 

- https://eslint.org/docs/latest/rules/no-multiple-empty-lines
- https://eslint.org/docs/latest/rules/eol-last

File: .prettierrc (hunk 3 above)
Comment (eerison) [reply .prettierrc:4]: I checked prettier, and it's optional,  there are others rules, but if you want to add , it's up to you

**Comment by eerison** [comment]
Don't forget to link this pr with issue in pr's description

**Comment by Willian1661** [comment]
> Don't forget to link this pr with issue in pr's description

oh true, i already did

File: .eslintrc.js (hunk 6 above)
Comment (Willian1661) [reply .eslintrc.js:46]: ok, i did 
i was reading about other rules, but i think for now we don't need much rules running through the code, don't we?

**Comment by Willian1661** [comment]
@eerison 
a changed extends:
"extends": ["eslint:recommended", "prettier"], => "now"
"extends": ["eslint:recommended", ".prettierrc"], => "before"
i was calling the file .prettierrc instead of the prettier by itself.

and 'cause of this Eslint wasn't running on codebase

File: .eslintrc.js (hunk 6 above)
Comment (Willian1661) [reply .eslintrc.js:46]: i added these rules into the script: 
"no-multiple-empty-lines": "error",
"eol-last": [
			"error", 
			"always"
		],
it's all done here with the rules

File: .eslintrc.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
+module.exports = {
+	"env": {
+		"browser": true,
+		"es2021": true
```
Comment (Willian1661) [inline .eslintrc.js:4]: i think i should change for "es2022", don't u think ?

File: .eslintrc.js (hunk 6 above)
Comment (eerison) [reply .eslintrc.js:46]: Then there is something wrong,  because your pipelines doesn't fail.

Is it working locally?

Comment (eerison) [reply .eslintrc.js:46]: @Willian1661 

When you going to use block code 

Use json blocks,  look into github markdown

**Comment by Willian1661** [comment]
> @eerison a changed extends: "extends": ["eslint:recommended", "prettier"], => "now" "extends": ["eslint:recommended", ".prettierrc"], => "before" i was calling the file .prettierrc instead of the prettier by itself.
> 
> and 'cause of this Eslint wasn't running on codebase

@eerison 
may you take a look if there's something missing/wrong ?

**Pull request merged**